react-basic-practice/src/components/Modal/index.tsx

```diff
@@ -8,36 +8,50 @@ import {
   ModalBody,
 } from "@chakra-ui/react";
 import Button from "../Button";
-import Form from "../Form";
 
 interface ModalProps {
   isOpen: boolean;
   onClose: () => void;
-  headerText?: string;
+  title: string;
+  onSubmit?: () => void;
+  children: React.ReactNode;
 }
 
 const Modal: React.FC<ModalProps> = ({
   isOpen,
   onClose,
-  headerText = "Add Customer",
+  title,
+  onSubmit,
+  children,
 }) => {
   return (
     <ChakraModal isOpen={isOpen} onClose={onClose}>
       <ModalOverlay />
       <ModalContent maxWidth="794px" padding="0 40px">
-        <ModalHeader padding="23px 0 18px 0" borderBottom="1px solid black" fontWeight="semibold" fontSize="lg" color="mirage">
-          {headerText}
+        <ModalHeader
+          padding="23px 0 18px 0"
+          borderBottom="1px solid black"
+          fontWeight="semibold"
+          fontSize="lg"
+          color="mirage"
+        >
+          {title}
         </ModalHeader>
-        <ModalBody padding="19px 0 0 0">
-          <Form />
-        </ModalBody>
+        <ModalBody padding="19px 0 0 0">{children}</ModalBody>
         <ModalFooter
           justifyContent="center"
           flexDirection="column"
           gap="13px"
           padding="45px 0 27px 0"
         >
-          <Button label="Create" variant="primary" size="sm" />
+          {onSubmit && (
+            <Button
+              label={title === "Edit Customer" ? "Save" : "Create"}
```

Comment:
pls add more type `isEdit` to check this condition. Don't check it with title here

Reply:
i done